### PR TITLE
Added a '--cache-only' feature for installation.

### PR DESCRIPTION
**Pull request by DigitalInBlue**
I added a "--cache-only" feature for installation.  This is needed because I have a use case where my development network is not connected to the Internet.  The machines that ARE connected to the Internet are not where I need to actually build.  I want to create a cache of all required packages, move it to the development network, and use the cached packages to build at that point.

### CODE REVIEW

**Comment by cbezault** [comment]
While I agree with some of the whitespace changes that you made please only make them as a drive-by on lines or sections that you actually have to touch for your feature, it makes it easier to look at the diff.

**Comment by DigitalInBlue** [comment]
Agree on the whitespace changes.  That was visual studio applying your .clang-format, not myself making manual changes.

**Comment by cbezault** [comment]
Hmmm, I also don't see why you can't just use the `export` command to get NuGet packages to use on your isolated machines.

**Comment by DigitalInBlue** [comment]
I need support for Windows and Linux, multiple compilers.  I also don't have a computer that is very good attached to the Internet.  Building on it is extremely painful.  We maintain a vcpkg cache on our disconnected network to build from on actual development machines.

**Comment by cbezault** [comment]
Okay, it wasn't immediately clear from the dirty diff but now I see that you don't actually build, you really just get all the source. In that case I like this but maybe I'll ask the other people on the team if they can think of a more intuitive name than `--cache-only`.

**Comment by DigitalInBlue** [comment]
Correct.  There is no building.  This feature just downloads all of the necessary packages into the ```Downloads``` folder.  We then take this folder to our development network and continue on.  

No pride in ownership.  This has been a great feature for our team and specific use case.

**Comment by cbezault** [comment]
And I think it's a useful feature that I'd like to see mainlined. Thanks for the PR!

File: toolsrc/include/vcpkg/build.h [hunk 1]
```diff
@@ -82,6 +82,12 @@ namespace vcpkg::Build
         YES
     };
 
+	enum class CacheOnly
```
Comment (vicroms) [inline toolsrc/include/vcpkg/build.h]: Use spaces for indentation instead of tabs.
This should be consistent with the rest of the project.

**Comment by DigitalInBlue** [comment]
I will update the formatting with your ```.clang-format```.  Do you want me to change the name of the flag as well?

**Comment by vicroms** [comment]
I would rather have a new command `vcpkg download <package>` instead of just adding more options on top of `vcpkg install`.

**Comment by DigitalInBlue** [comment]
```vcpkg download``` sounds reasonable to me.  I'll work that and get the change pushed back up.  Do you want a new pull request or just an update to this one?

**Comment by cbezault** [comment]
Just update this one.

**Comment by DigitalInBlue** [comment]
This PR is failing and I've got a few things to clean up.  I am going to close this PR, clean up my fork, and initiate a new PR.  I will reference this PR in my new PR's comments.

**Pull request closed without merging**